mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/UserReindexingTask.java

```diff
@@ -25,13 +25,41 @@
 
 import org.apache.james.core.User;
 import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.server.task.json.dto.TaskDTO;
 import org.apache.james.task.Task;
 import org.apache.james.task.TaskExecutionDetails;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+
 public class UserReindexingTask implements Task {
 
     public static final String USER_RE_INDEXING = "userReIndexing";
 
+    public static class UserReindexingTaskDTO implements TaskDTO {
+
+        private final String type;
+        private final String username;
+
+        private UserReindexingTaskDTO(@JsonProperty("type") String type, @JsonProperty("username") String username) {
+            this.type = type;
+            this.username = username;
+        }
+
+        @Override
+        public String getType() {
+            return type;
+        }
+
+        public String getUsername() {
+            return username;
+        }
+
+        public static UserReindexingTaskDTO of(String type, UserReindexingTask task) {
```

Comment:
static method ordering

Reply:
Ok, but i must admin i don't understand this rule.
I think it's much clearer to have the field and methods of the instance. Then at the end the static part. It enable to see at first sight what the class is meant to do.

Like in scala you declare the class first then the companion object after in the same file.

Reply:
I'm rather ok with you but:
1/ static first is the de-facto standard in Java
2/ if we change ordering, we have to change the checkstyle rule and refactor the whole codebase

Does it worth it?

Reply:
There is no checkstyle on this yet. So it does not bind us to refactor the whole codebase.
By the way i followed the current rules in this PR.

Reply:
> There is no checkstyle on this yet. So it does not bind us to refactor the whole codebase.

Well, if we decide a change of our code guidelines, we have to apply it globally. The worst situation is a mix of two styles.